src/server/models/Reading.js

```diff
@@ -206,6 +210,34 @@ class Reading {
 		return barchartReadingsByMeterID;
 	}
 
+	/**
+	 * Gets barchart readings for every group across the given time interval for a duration.
+	 *
+	 * Compressed readings are in kilowatts.
+	 * @param groupIDs an array of ids for groups whose points are being compressed
+	 * @param duration A moment time duration over which to sum the readings
+	 * @param fromTimestamp An optional start point for the beginning of the entire time range.
+	 * @param toTimestamp An optional end point for the end of the entire time range.
+	 * @param conn the connection to use. Defaults to the default database connection.
+	 * @return {Promise<object<int, array<{reading_sum: number, start_timestamp: Date, end_timestamp: Date}>>>}
+	 */
+	static async getGroupBarchartReadings(groupIDs, duration, fromTimestamp = null, toTimestamp = null, conn = db) {
+		const allBarchartReadings = await conn.func('barchart_group_readings', [groupIDs, duration, fromTimestamp || '-infinity', toTimestamp || 'infinity']);
+		// Separate the result rows by meter_id and return a nested object.
+		const barchartReadingsByGroupID = {};
+		for (const row of allBarchartReadings) {
+			console.log(JSON.stringify(row));
```

Comment:
Remove console.log

Reply:
These eventually got fixed in 0e76608b24f7bf1bc4b18d42e73eff2a1e0b5381.